### PR TITLE
Merge latest updates to 'main' branch

### PR DESCRIPTION
**Pull request by YuelinXin**
1. Update links in templates
2. Update code_of_conduct
3. Add new functionality: bookmark

### CODE REVIEW

**Comment by YuelinXin** [comment]
You can review my code and merge it after you think it is AOK @Bilgin-L

**Comment by Bilgin-L** [comment]
Got it. I will review it later.



发自我的iPhone


------------------ Original ------------------
From: Yuelin Xin ***@***.***&gt;
Date: Tue,Aug 2,2022 9:22 PM
To: MiracleFactory/MF-Project-Official-Website ***@***.***&gt;
Cc: Bilgin ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [MiracleFactory/MF-Project-Official-Website] Merge latest updates to 'main' branch (PR #3)





 
You can review my code and merge it after you think it is AOK @Bilgin-L

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Contributor Covenant Code of Conduct
 
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-2.1-4baaaa.svg)](code_of_conduct.md)
```
Comment (YuelinXin) [inline CODE_OF_CONDUCT.md:3]: just adding a badge

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -60,7 +62,7 @@ representative at an online or offline event.
 
 Instances of abusive, harassing, or otherwise unacceptable behavior may be
 reported to the community leaders responsible for enforcement at
-[INSERT CONTACT METHOD].
+[service@miraclefactory.co].
```
Comment (YuelinXin) [inline CODE_OF_CONDUCT.md:65]: replace the placeholder with our contact info

File: README.md [hunk 3]
```diff
@@ -19,7 +19,7 @@ You can make suggestions and contributions on the website's:
 
 ### Repository Maintenance 🔨
 **ANY FORM** of content / style change should be submitted via Pull Requests.  
-If you wish to make adjustments to this repo, please commit your change to branch **dev** first, then create a PR to merge it into our main branch.  
+If you wish to make adjustments to this repo, please fork this repo and commit your change to branch **main** in your fork first, then create a PR to merge it into our **main** branch.
```
Comment (YuelinXin) [inline README.md:22]: clarify our maintenance policy.

File: application/static/CSS/main.css [hunk 4]
```diff
@@ -122,6 +122,44 @@ svg
...
+    font-size: 13px;
+    line-height: 27px;
+    font-weight: bold;
+}
+
+.dialog-button-no:hover
+{
+    background-color: rgb(180, 63, 56);
+}
+
```
Comment (YuelinXin) [inline application/static/CSS/main.css:162]: add CSS for bookmark, see element `id="bookmark"` in `/application/templates/index.html`

Comment (YuelinXin) [reply application/static/CSS/main.css:162]: sorry, these are the CSS for the buttons in that element, not the element itself.

File: application/static/CSS/main.css [hunk 5]
```diff
@@ -200,6 +238,29 @@ svg
...
+    box-shadow: 0px 0px 15px rgba(0, 0, 0, 0.2);
+    transform: translateX(calc(100% - 28px));
+    transition: all 0.4s ease-in-out;
+}
+
+#bookmark:hover
+{
+    transform: translateX(0);
+}
+
```
Comment (YuelinXin) [inline application/static/CSS/main.css:263]: these are the CSS for the "bookmark" element in `index.html`

File: application/static/CSS/main.css [hunk 6]
```diff
@@ -579,6 +640,12 @@ textarea:focus
         background-color: rgb(240, 240, 240);
     }
 
+    #bookmark
+    {
+        background-color: rgb(55, 59, 64);
+        border-color: rgb(122, 123, 127);
+    }
+
```
Comment (YuelinXin) [inline application/static/CSS/main.css:648]: "bookmark" in dark mode

File: application/static/CSS/main.css [hunk 7]
```diff
@@ -778,6 +845,11 @@ textarea:focus
 
 @media screen and (max-width: 400px) {
 
+    #bookmark
+    {
+        display: none;
+    }
+
```
Comment (YuelinXin) [inline application/static/CSS/main.css:852]: hide "bookmark" in small displays like a phone

File: application/static/JS/basic.js [hunk 8]
```diff
@@ -65,3 +65,8 @@ function closeDialog()
     document.getElementById("dialog-contact-bk").style.display = "none";
     document.body.style.overflowY = "auto";
 }
+
+function closeBookmark()
+{
+    document.getElementById("bookmark").style.opacity = 0;
+}
```
Comment (YuelinXin) [inline application/static/JS/basic.js:72]: a simple JS function for dismissing the bookmark

**Comment by Bilgin-L** [comment]
Looks GREAT

**Review by Bilgin-L — APPROVED** [review]
New bookmark is pretty nice.

**Pull request merged**